isvcs/zk.go

```diff
@@ -241,13 +307,70 @@ func BuildZookeeperInstance(instanceID int) dao.RunningService{
 	return *newInst
 }
 
-func GetZookeeperInstances() []dao.RunningService {
+func GetZooKeeperRunningInstances() []dao.RunningService {
 	instances := []dao.RunningService{}
 
 	// loop through and create an instance for each IP we have.
-	for count, _ := range zkConnectStrings {
-		instances = append(instances, BuildZookeeperInstance(count))
+	for _, key := range GetZooKeeperKeys() {
+		instances = append(instances, BuildZooKeeperRunningInstance(key.InstanceID))
+	}
+
+	return instances
+}
+
+type ZooKeeperInstance struct {
+	dao.RunningService
+	IP    string
+	Port  int
+	Stats ZooKeeperServerStats
+}
+
+func GetZooKeeperInstances() []ZooKeeperInstance {
+	instances := []ZooKeeperInstance{}
+
+	// loop through and create an instance for each IP we have.
+	for _, key := range GetZooKeeperKeys() {
+		logger := log.WithField("connection", key.Connection)
+
+		tokens := strings.SplitN(key.Connection, ":", 2)
+		if len(tokens) < 2 {
+			logger.Error("Unable to parse ZooKeeper connection")
+			continue
+		}
+		ip := tokens[0]
+
+		port, err := strconv.Atoi(strings.TrimSpace(tokens[1]))
+		if err != nil {
+			logger.Info("Unable to set ZooKeeper port for stats")
+		}
+
+		instances = append(instances, ZooKeeperInstance{
+			BuildZooKeeperRunningInstance(key.InstanceID),
+			ip,
+			port,
+			GetZooKeeperServerStats(key.InstanceID, key.Connection)})
 	}
 
 	return instances
 }
+
+type ZooKeeperKey struct {
+	InstanceID int
+	Connection string
+}
+
+func GetZooKeeperKeys() []ZooKeeperKey {
```

Comment:
This more or less duplicates the logic at lines 38-49.  Please rework something so all of the logic in this file operates off a single list of ZK instances (or single function that builds a list)